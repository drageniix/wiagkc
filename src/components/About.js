import React from "react"

export const About = ({ data }) => (
    <section className="about" id="about">
        <h2 className="section__title">{data.title}</h2>
        <p className="about__content" dangerouslySetInnerHTML={{ __html: data.content}}></p>
    </section>
)

export default About