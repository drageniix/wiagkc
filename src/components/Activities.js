import React from "react"
import PropTypes from "prop-types"
import { connect } from "react-redux"
import SVG from "./SVG"

export const Activities = ({ data }) => (
    <section className="activities" id="activities">
        <section className="activities__box">
            {data.types.map((item, index) => (
                <div key={index} className="activities__activity">
                    <div className="activities__activity--icon">
                        <SVG svg={item.icon}/>
                    </div>
                    <h4 className="activities__activity--header">{item.type}</h4>
                    <ul className="activities__activity--exampleList">
                        {item.examples.map((ex, exIndex) => (
                            <li key={exIndex} className="activities__activity--example">
                                {ex}
                            </li>
                        ))}
                    </ul>
                </div>
            ))}
        </section>
    </section>
)

const mapStateToProps = state => ({
    data: state.activities
})

Activities.propTypes = {
    data: PropTypes.object.isRequired
}
export default connect(mapStateToProps)(Activities)