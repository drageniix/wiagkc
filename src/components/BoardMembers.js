import React from "react"

export const BoardMembers = ({ data }) => (
  <section id="boardMembers" className="boardMembers">
    <h2 className="section__title">{data.title}</h2>
    <div className="boardMembers__members">
      {data.officers.map((item, index) => (
        <div key={index} className="boardMembers__member">
            <h4 className="boardMembers__member--title">{item.title}</h4>
            <span className="boardMembers__member--name">{item.name}</span>
            <span> </span>
            <span className="boardMembers__member--credential">{item.credentials}</span>
          </div>
      ))}
    </div>
  </section>
)

export default BoardMembers