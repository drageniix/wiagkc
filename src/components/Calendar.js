import React from "react"
import moment from 'moment'

export const Calendar = ({ data }) => (
    <section id="calendar" className="calendar">
        <section className="calendar__container">
            <h2 className="section__title">{data.title}</h2>
            <div className="calendar__events">
                {data.events.map((item, index) => (
                    <div key={index} className="calendar__event">
                        <div className="calendar__event--date">{moment.unix(item.date).format("MMM Do")}</div>
                        <div className="calendar__event--details">
                            {item.bold ? 
                                <strong><p className={item.class} dangerouslySetInnerHTML={{ __html: item.event }}></p></strong> : 
                                <p className={item.class}>{item.event}</p>
                            }
                            <p>{item.details}</p>
                        </div>
                    </div>
                ))}
            </div>
        </section>
    </section>
)

export default Calendar