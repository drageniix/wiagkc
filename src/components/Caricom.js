import React from "react"
import ResponsiveImage from "./ResponsiveImage"
import { connect } from "react-redux"
import PropTypes from "prop-types"

export class Caricom extends React.Component {
    state = {modal : ""}

    openModal = () => {
        this.setState({modal : this.props.data})
    }

    closeModal = event => {
        event.preventDefault()
        if (event.target === event.currentTarget) {
            this.setState({ modal: "" })
        }
    }
    
    render(){
        return (
            <section id="caricom" className="caricom">
                { this.state.modal ? <CaricomModal data={this.state.modal} close={this.closeModal} /> : undefined }
                <figure className="caricom__map">    
                    <ResponsiveImage className="caricom__map--image" image={this.props.data.map.image} />
                    <figcaption className="caricom__map--caption">{this.props.data.map.caption}</figcaption>
                </figure>
                <h2 className="section__title">{this.props.data.title}</h2>
                <p className="caricom__text">{this.props.data.blurb} <a onClick={this.openModal}>Learn More...</a></p> 
            </section>
        )
    }
}

Caricom.propTypes = {
    data: PropTypes.object.isRequired
}

const path = location.pathname.slice(0, location.pathname.lastIndexOf("/"))
export const CaricomModal = ({data, close}) => (
    <div className="modal" onClick={close}>
        <div className="modal__content">
            <h2 className="section__title">{data.title}</h2>
            <div className="modal__text">
                {data.more.description.map((item, index) => (
                    <p key={index}>
                        {item}
                    </p>
                ))}
                <div>
                    <a href={path + "/" + data.map.image.src}
                        rel="noopener noreferrer" target="_blank">
                        <ResponsiveImage className="modal__image" image={data.map.image} />
                    </a>
                    <caption className="modal__caption">{data.map.caption}</caption>
                </div>
            </div>
        </div>
    </div>
)

const mapStateToProps = state => ({
    data: state.caricom
})

CaricomModal.propTypes = {
    data: PropTypes.object.isRequired,
    close: PropTypes.object.isRequired
}

export default connect(mapStateToProps)(Caricom)