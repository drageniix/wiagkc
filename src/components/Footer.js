import React from 'react'
import ResponsiveImage from './ResponsiveImage'

const Footer = ({ data }) => (
    <section className="footer">
        <div className="footer__logobox">
            <ResponsiveImage className="footer__logo" image={data.logo} />
            <h2 className="section__title">{data.title}</h2>
        </div>
        <div className="footer__data">
            <p>{data.email}</p>
            <p>{data.phone}</p>
            <p>{data.address}</p>
        </div>
    </section>
)

export default Footer