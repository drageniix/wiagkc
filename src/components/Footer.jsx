import React from 'react';
import ResponsiveImage from 'responsive-json-webpack-plugin/react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const Footer = ({ data }) => (
    <footer className="section footer" id="footer">
        <div className="footer__logobox">
            <ResponsiveImage
                className="footer__logobox--logo"
                image={data.logo}
            />
            <h2 className="section__title">{data.title}</h2>
        </div>
        <div className="footer__data">
            <p>
                <a href={`https://twitter.com/${data.twitter}`}>
                    @{data.twitter}
                </a>
            </p>
            <p>
                <a href={'tel:' + data.phone.replace('-', '')}>{data.phone}</a>
            </p>
            <p>
                <a href={'mailto:' + data.email}>{data.email}</a>
            </p>
            <p>
                <a href={data.geo} target="_blank" rel="noopener noreferrer">
                    {data.address}
                </a>
            </p>
        </div>
    </footer>
);

const mapStateToProps = state => ({
    data: state.common.marginals
});

Footer.propTypes = {
    data: PropTypes.object.isRequired
};

export default connect(mapStateToProps)(Footer);
