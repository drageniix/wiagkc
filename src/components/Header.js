import React from "react"

const Header = ({ data }) => (
    <header className="header" id="top">
        <div className="header__hero">
            <h1 className="header__hero--title">{data.title}</h1>
            <h2 className="header__hero--subtitle">{data.subtitle}</h2>
            <h3 className="header__hero--tagline">{data.tagline}</h3>
        </div>
    </header>
)

export default Header
