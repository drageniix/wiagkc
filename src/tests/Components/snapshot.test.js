import React from 'react'
import { About } from '../../components/About'
import { Activities } from '../../components/Activities'
import { BoardMembers } from '../../components/BoardMembers'
import { Calendar } from '../../components/Calendar'
import { Caricom, CaricomModal } from '../../components/Caricom'
import { Footer } from '../../components/Footer'
import { Header } from '../../components/Header'
import { Membership } from '../../components/Membership'
import { Navigation } from '../../components/Navigation'
import { NextMeeting } from '../../components/NextMeeting'
import { Photos } from '../../components/Photos'


test('Marginals', () => {
    const wrapperFooter = shallow(
        <Footer data={data.marginals} />
    )
    expect(wrapperFooter).toMatchSnapshot()


    const wrapperHeader = shallow(
        <Header data={data.marginals} />
    )
    expect(wrapperHeader).toMatchSnapshot()

    const wrapperNavigation = shallow(
        <Navigation data={data.marginals} />
    )
    expect(wrapperNavigation).toMatchSnapshot()
})

test('About', () => {
    const wrapper = shallow(
        <About data={data.about} />
    )
    expect(wrapper).toMatchSnapshot()
})

test('Caricom', () => {
    const wrapper = shallow(
        <Caricom data={data.caricom} />
    )
    expect(wrapper).toMatchSnapshot()
})

test('Activities', () => {
    const wrapper = shallow(
        <Activities data={data.activities} />
    )
    expect(wrapper).toMatchSnapshot()
})


test('Calendar', () => {
    const wrapperCalendar = shallow(
        <Calendar data={data.calendar} />
    )
    expect(wrapperCalendar).toMatchSnapshot()

    const wrapperNext = shallow(
        <NextMeeting data={data.calendar} />
    )
    expect(wrapperNext).toMatchSnapshot()

})

test('Membership', () => {
    const wrapper = shallow(
        <Membership data={data.membership} />
    )
    expect(wrapper).toMatchSnapshot()
})

test('Photos', () => {
    const wrapper = shallow(
        <Photos data={data.photos} />
    )
    expect(wrapper).toMatchSnapshot()
})

test('BoardMembers', () => {
    const wrapper = shallow(
        <BoardMembers data={data.boardMembers} />
    )
    expect(wrapper).toMatchSnapshot()
})