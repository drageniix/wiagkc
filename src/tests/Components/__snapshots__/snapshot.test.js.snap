// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About 1`] = `
<section
  className="about"
  id="about"
>
  <h2
    className="section__title"
  >
    Our Mission
  </h2>
  <p
    className="about__content"
    dangerouslySetInnerHTML={
      Object {
        "__html": "The <strong>West Indian Association of Greater Kansas City (WIA-KC)</strong> is a 501(c)(3) non-profit organization with the objective to promote educational, youth, literary, charitable and cultural activities with the intent to maintain, preserve, and foster the cultural heritage of the people of the West Indies; and to encourage friendship among individuals who are interested in West Indian culture and to promote friendship and understanding among the West Indian community and with people of other cultural backgrounds.",
      }
    }
  />
</section>
`;

exports[`Activities 1`] = `
<section
  className="activities"
  id="activities"
>
  <section
    className="activities__box"
  >
    <div
      className="activities__activity"
      key="0"
    >
      <div
        className="activities__activity--icon"
      >
        <div
          className="icon--teacher"
        />
      </div>
      <h4
        className="activities__activity--header"
      >
        Education
      </h4>
      <ul
        className="activities__activity--exampleList"
      >
        <li
          className="activities__activity--example"
          key="0"
        >
          STEM Opportunities
        </li>
        <li
          className="activities__activity--example"
          key="1"
        >
          West Indian Cooking
        </li>
        <li
          className="activities__activity--example"
          key="2"
        >
          Money Management
        </li>
      </ul>
    </div>
    <div
      className="activities__activity"
      key="1"
    >
      <div
        className="activities__activity--icon"
      >
        <div
          className="icon--help"
        />
      </div>
      <h4
        className="activities__activity--header"
      >
        Community Service
      </h4>
      <ul
        className="activities__activity--exampleList"
      >
        <li
          className="activities__activity--example"
          key="0"
        >
          Food Drive
        </li>
        <li
          className="activities__activity--example"
          key="1"
        >
          Clothing Drive
        </li>
        <li
          className="activities__activity--example"
          key="2"
        >
          Guest Speakers
        </li>
      </ul>
    </div>
    <div
      className="activities__activity"
      key="2"
    >
      <div
        className="activities__activity--icon"
      >
        <div
          className="icon--beach-house"
        />
      </div>
      <h4
        className="activities__activity--header"
      >
        Cultural Events
      </h4>
      <ul
        className="activities__activity--exampleList"
      >
        <li
          className="activities__activity--example"
          key="0"
        >
          Jerk Festival
        </li>
        <li
          className="activities__activity--example"
          key="1"
        >
          Family Picnics
        </li>
        <li
          className="activities__activity--example"
          key="2"
        >
          Book Clubs
        </li>
      </ul>
    </div>
    <div
      className="activities__activity"
      key="3"
    >
      <div
        className="activities__activity--icon"
      >
        <div
          className="icon--customer"
        />
      </div>
      <h4
        className="activities__activity--header"
      >
        Mentorship
      </h4>
      <ul
        className="activities__activity--exampleList"
      >
        <li
          className="activities__activity--example"
          key="0"
        >
          Job Shadowing
        </li>
        <li
          className="activities__activity--example"
          key="1"
        >
          Business Networking
        </li>
      </ul>
    </div>
    <div
      className="activities__activity"
      key="4"
    >
      <div
        className="activities__activity--icon"
      >
        <div
          className="icon--partnership"
        />
      </div>
      <h4
        className="activities__activity--header"
      >
        Strategic Partnerships
      </h4>
      <ul
        className="activities__activity--exampleList"
      >
        <li
          className="activities__activity--example"
          key="0"
        >
          Sponsorships
        </li>
        <li
          className="activities__activity--example"
          key="1"
        >
          West Indian Associations
        </li>
        <li
          className="activities__activity--example"
          key="2"
        >
          Cultural Heritage Organizations
        </li>
      </ul>
    </div>
  </section>
</section>
`;

exports[`BoardMembers 1`] = `
<section
  className="boardMembers"
  id="boardMembers"
>
  <h2
    className="section__title"
  >
    Board Members
  </h2>
  <div
    className="boardMembers__members"
  >
    <div
      className="boardMembers__member"
      key="0"
    >
      <h4
        className="boardMembers__member--title"
      >
        President
      </h4>
      <span
        className="boardMembers__member--name"
      >
        Dr. Janice Barrow
      </span>
      <span>
         
      </span>
      <span
        className="boardMembers__member--credential"
      >
        Kansas State University
      </span>
    </div>
    <div
      className="boardMembers__member"
      key="1"
    >
      <h4
        className="boardMembers__member--title"
      >
        Founder and Vice-President – Public Relations and Membership,
      </h4>
      <span
        className="boardMembers__member--name"
      >
        Mrs. Hyacinth Henry
      </span>
      <span>
         
      </span>
      <span
        className="boardMembers__member--credential"
      >
        Entrepreneur and Community Activist
      </span>
    </div>
    <div
      className="boardMembers__member"
      key="2"
    >
      <h4
        className="boardMembers__member--title"
      >
        Secretary
      </h4>
      <span
        className="boardMembers__member--name"
      >
        Keisha Jones
      </span>
      <span>
         
      </span>
      <span
        className="boardMembers__member--credential"
      >
        RN, CPHIMS, Manager, Clinical Consultant Knowledge Lead – Consulting, Cerner
      </span>
    </div>
    <div
      className="boardMembers__member"
      key="3"
    >
      <h4
        className="boardMembers__member--title"
      >
        Assistant Secretary
      </h4>
      <span
        className="boardMembers__member--name"
      >
        Therese Dublin
      </span>
      <span>
         
      </span>
      <span
        className="boardMembers__member--credential"
      >
        RN
      </span>
    </div>
    <div
      className="boardMembers__member"
      key="4"
    >
      <h4
        className="boardMembers__member--title"
      >
        Treasurer
      </h4>
      <span
        className="boardMembers__member--name"
      >
        Keithly Lake
      </span>
      <span>
         
      </span>
      <span
        className="boardMembers__member--credential"
      >
        Attorney-at-Law
      </span>
    </div>
    <div
      className="boardMembers__member"
      key="5"
    >
      <h4
        className="boardMembers__member--title"
      >
        Legal Advisor
      </h4>
      <span
        className="boardMembers__member--name"
      >
        Keithly Lake
      </span>
      <span>
         
      </span>
      <span
        className="boardMembers__member--credential"
      >
        Attorney-at-Law
      </span>
    </div>
    <div
      className="boardMembers__member"
      key="6"
    >
      <h4
        className="boardMembers__member--title"
      >
        Legal Advisor
      </h4>
      <span
        className="boardMembers__member--name"
      >
        Stacy Lake
      </span>
      <span>
         
      </span>
      <span
        className="boardMembers__member--credential"
      >
        Attorney-at-Law, Lake Firm, LLC
      </span>
    </div>
  </div>
</section>
`;

exports[`Calendar 1`] = `
<section
  className="calendar"
  id="calendar"
>
  <section
    className="calendar__container"
  >
    <h2
      className="section__title"
    >
      Planned Activities
    </h2>
    <div
      className="calendar__events"
    >
      <div
        className="calendar__event"
        key="0"
      >
        <div
          className="calendar__event--date"
        >
          May 17th
        </div>
        <div
          className="calendar__event--details"
        >
          <p>
            ...Loading Events
          </p>
          <p />
        </div>
      </div>
    </div>
  </section>
</section>
`;

exports[`Calendar 2`] = `
<section
  className="nextMeeting"
  id="nextMeeting"
>
  <p
    className="nextMeeting__title"
  >
    ...Loading Events
  </p>
  <p
    className="nextMeeting__date"
  >
    Tuesday, May 17th 2033, 10:33 pm
  </p>
</section>
`;

exports[`Caricom 1`] = `
<section
  className="caricom"
  id="caricom"
>
  <figure
    className="caricom__map"
  >
    <Connect(ResponsiveImage)
      className="caricom__map--image"
      image={
        Object {
          "alt": "Map of the Caricom Members",
          "src": "caricom.png",
        }
      }
    />
    <figcaption
      className="caricom__map--caption"
    >
      Map of Caricom Members
    </figcaption>
  </figure>
  <h2
    className="section__title"
  >
    The West Indies
  </h2>
  <p
    className="caricom__text"
  >
    Countries of the West Indies (an area of the Caribbean) share a common cultural history created by European colonization of the region, as well as the significant presence of people of African descent. The region is not only informally unified by its culture, heritage, and sports, but is also formally unified, in part, by organizations such as CARICOM.
     
    <a
      onClick={[Function]}
    >
      Learn More...
    </a>
  </p>
</section>
`;

exports[`Marginals 1`] = `
<section
  className="footer"
  id="footer"
>
  <div
    className="footer__logobox"
  >
    <Connect(ResponsiveImage)
      className="footer__logobox--logo"
      image={
        Object {
          "alt": "WIA-GKC Logo",
          "src": "logo.png",
        }
      }
    />
    <h2
      className="section__title"
    >
      West Indian Association
    </h2>
  </div>
  <div
    className="footer__data"
  >
    <p>
      <a
        href="https://twitter.com/wiagkc"
      >
        @
        wiagkc
      </a>
    </p>
    <p>
      <a
        href="mailto:contact@wiakc.org"
      >
        contact@wiakc.org
      </a>
    </p>
    <p>
      <a
        href="tel:417986-6144"
      >
        417-986-6144
      </a>
    </p>
    <p>
      <a
        href="geo:39.070497, -94.574570"
        target="_blank"
      >
        817 East 31st Street, Kansas City, MO 64109
      </a>
    </p>
  </div>
</section>
`;

exports[`Marginals 2`] = `
<header
  className="header"
  id="top"
>
  <div
    className="header__hero"
  >
    <h1
      className="header__hero--title"
    >
      West Indian Association
    </h1>
    <h2
      className="header__hero--subtitle"
    >
      of Greater Kansas City
    </h2>
    <h3
      className="header__hero--tagline"
    >
      Sharing the West Indian Culture in Kansas City
    </h3>
  </div>
</header>
`;

exports[`Marginals 3`] = `
<nav
  className="navigation"
>
  <Connect(ResponsiveImage)
    className="navigation__logo"
    image={
      Object {
        "alt": "WIA-GKC Logo",
        "src": "logo.png",
      }
    }
  />
  <span
    className="navigation__title"
  >
    WIA-KC
  </span>
  <span
    className="navigation__links"
  >
    <a
      className="navigation__links--link"
      href="#top"
      key="0"
    >
      Home
    </a>
    <a
      className="navigation__links--link"
      href="#about"
      key="1"
    >
      About Us
    </a>
    <a
      className="navigation__links--link"
      href="#calendar"
      key="2"
    >
      Activities
    </a>
    <a
      className="navigation__links--link"
      href="#membership"
      key="3"
    >
      Membership
    </a>
  </span>
</nav>
`;

exports[`Membership 1`] = `
<section
  className="membership"
  id="membership"
>
  <div
    className="membership__container"
  >
    <div
      className="membership__frequency"
    >
      <h2
        className="section__title"
      >
        Meetings
      </h2>
      <p
        className="membership__frequency--main"
      >
        <strong>
           🏝️ 
          2nd Saturday of Each Month
           
        </strong>
      </p>
      <p
        className="membership__frequency--exception"
      />
    </div>
    <p
      className="membership__detail"
    >
      There is something for everyone as we invite you to join our association, volunteer, and/or enjoy the many activities planned.
    </p>
    <p
      className="membership__detail"
    >
      Please attend our meetings and share ideas for other projects or issues you feel are important in the West Indian community. If unable to attend, please read a recap of meeting minutes from last month. Please refer other potential members.
    </p>
  </div>
</section>
`;

exports[`Photos 1`] = `
<section
  className="photos"
  id="photos"
>
  <section
    className="photos__images"
  >
    <Connect(ResponsiveImage)
      image={
        Object {
          "alt": "2018 Events 1",
          "src": "jerkfest-800.jpg",
        }
      }
      key="0"
    />
    <Connect(ResponsiveImage)
      image={
        Object {
          "alt": "2018 Events 2",
          "src": "picnic-800.jpg",
        }
      }
      key="1"
    />
  </section>
</section>
`;
